src/dbus/api/secret.rs

```diff
@@ -26,12 +26,9 @@ impl<'a> Secret<'a> {
         secret: &[u8],
         content_type: &str,
     ) -> Self {
-        let (parameters, value) = match algorithm.as_ref() {
-            Algorithm::Plain => (vec![], secret.to_vec()),
-            Algorithm::Dh(_blob) => {
-                // See https://github.com/hwchen/secret-service-rs/blob/d6aaa774f0ec504ff5f26662279e07175b8ef111/src/util.rs#L52
-                unimplemented!()
-            }
+        let (value, parameters) = match algorithm.as_ref() {
+            Algorithm::Plain => (secret.to_vec(), vec![]),
+            Algorithm::Dh(aes_key) => utils::encrypt(secret, aes_key).unwrap(),
```

Comment:
The key should no longer be stored inside Dh variant, I would also maybe just rename it to Encrypted  and we could document what encryption it uses in the future. The key then should be generated and stored as part of src/dbus/service.rs and passed to the src::api::Session when created.

Reply:
Renamed to Encrypted, without a inner field. With this one can drop all the `Arc`s required. The key should be retrievable from the session right?

Reply:
I moved it to src/dbus, so it probably needs a rebase here